lib/rtsp.c

```diff
@@ -822,6 +861,49 @@ CURLcode Curl_rtsp_parseheader(struct Curl_easy *data, char *header)
       (data->set.str[STRING_RTSP_SESSION_ID])[idlen] = '\0';
     }
   }
+  else if(checkprefix("Transport:", header)) {
+    /* e.g.: 'Transport: RTP/AVP/TCP;unicast;interleaved=5-6' */
+    char *start;
+    char *end;
+    start = header + 10;
+    while(*start) {
+      while(*start && *start != ';' && ISBLANK(*start) )
+        start++;
+      end = start;
+      while(*end && *end != ';')
+        end++;
+      if(checkprefix("interleaved=", start)) {
+        long chan1, chan2, chan;
+        char *endp;
+        char *p = start + 12;
+        chan1 = strtol(p, &endp, 10);
+        if(p != endp && chan1 >= 0 && chan1 <= 255) {
+          unsigned char *rtp_channel_mask = data->state.rtp_channel_mask;
+          chan2 = chan1;
+          if(*endp == '-') {
+            p = endp + 1;
+            chan2 = strtol(p, &endp, 10);
+            if(p == endp || chan2 < 0 || chan2 > 255) {
+              failf(data, "Unable to read the interleaved parameter from "
+                    "Transport header: [%s]", header);
```

Comment:
Shouldn't this return error when the data is clearly invalid?

Reply:
If this return error, do we increase the probability that the request fails? After all, this is just to check the validity of the data. If we continue and therefore missing some channel information, we may therefore discard the data of these channel when RECEIVE, Is this a little more acceptable than the SETUP request failure?